packages/client/foreman_ygg_worker/foreman_ygg_worker.spec

```diff
@@ -0,0 +1,62 @@
+%define debug_package %{nil}
+
+# https://github.com/theforeman/foreman_ygg_worker
+
+%global repo_orgname theforeman
+%global repo_name foreman_ygg_worker
+%global yggdrasil_libexecdir %{_libexecdir}/yggdrasil
+
+%global goipath         github.com/%{repo_orgname}/%{repo_name}
+
+%if 0%{?rhel} > 7 && ! 0%{?fedora}
+%define gobuild(o:) \
+        go build -buildmode pie -compiler gc -tags="rpm_crashtraceback libtrust_openssl ${BUILDTAGS:-}" -ldflags "${LDFLAGS:-} -linkmode=external -compressdwarf=false -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -extldflags '%__global_ldflags'" -a -v %{?**};
+%else
+%if ! 0%{?gobuild:1}
+%define gobuild(o:) GO111MODULE=off go build -buildmode pie -compiler gc -tags="rpm_crashtraceback ${BUILDTAGS:-}" -ldflags "${LDFLAGS:-} -linkmode=external -B 0x$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \\n') -extldflags '-Wl,-z,relro -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '" -a -v %{?**};
+%endif
+%endif
```

Comment:
what is this for?

Reply:
IIRC el8 and newer ship a `gobuild` macro which can be used to build hardened go binaries. EL7 doesn't have this macro and this chunk is a reimplementation for it, borrowed from here http://pkgs.devel.redhat.com/cgit/rpms/podman/tree/podman.spec?h=stream-container-tools-rhel8

Reply:
fascinating (in a bad way, but you're just a messenger here)